main.js

```diff
@@ -686,6 +686,35 @@ function _select_Gateway() { // Check for Gateway used to
     }
 
     /*
+     * Helper function to add "Mass Gathering" profile
+     */
+    function addMassProfile(prof, T1MassGat, T1MassRef, T1Gat, T1Ref, T2MassGat, T2MassRef, T2Gat, T2Ref, T3MassGat, T3MassRef, T3Gat, T3Ref, T4MassGat, T4MassRef, T4Gat, T4Ref)

```

Comment:
I kinda agree to some extent, because I admit that I did not look at the function enough to fully understand everything it does. But I am going to guess that this function would save on line space, but I think at the expense of simplicity. I have no problem with the profiles being longer, when they are easier to read and understand at a glance, without having to understand much if anything about the function that was used to create them.